sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionAsyncDatabase.java

```diff
@@ -82,6 +82,13 @@ public CosmosAsyncClientEncryptionKey getClientEncryptionKey(String id) {
 
         EncryptionKeyStoreProvider encryptionKeyStoreProvider =
             this.cosmosEncryptionAsyncClient.getEncryptionKeyStoreProvider();
+
+        if (!encryptionKeyStoreProvider.getProviderName().equals(encryptionKeyWrapMetadata.getType())) {
+            throw new IllegalArgumentException("The EncryptionKeyWrapMetadata Type value does not match with the " +
```

Comment:
in similar scenario, does non Cosmos Exception type reach public surface area in the DotNet SDK?

Reply:
Yes .NET has ArgumentException as well. To me this is correct and it should be non cosmos as it is validation on argument passed by user.

Reply:
how would it work on the async API? do we do validation upfront? or do we return a `Mono.of(IllegalArgException)`

Reply:
This validation is om async API itself which is doing validation upfront .